lib/travis/api/v3/queries/log.rb

```diff
@@ -9,6 +9,8 @@ def find(job)
       #if the log has been archived, go to s3
       if log.archived_at
         content = fetch.get(prefix).try(:body)
```

Comment:
I've added a check here for nil and included raising an error.
Also added a spec below.